packages/plugin-ext/src/plugin/command-registry.ts

```diff
@@ -104,8 +104,20 @@ export class CommandRegistryImpl implements CommandRegistryExt {
             return this.executeLocalCommand(id, ...args);
         } else if (KnownCommands.mapped(id)) {
             // Using the KnownCommand exclusions, convert the commands manually
-            return KnownCommands.map(id, args, (mappedId: string, mappedArgs: any[] | undefined) =>
-                this.proxy.$executeCommand(mappedId, ...mappedArgs));
+            return KnownCommands.map(id, args, (mappedId: string, mappedArgs: any[] | undefined, mappedResult: KnownCommands.ConversionFunction) => {
+                const mr: KnownCommands.ConversionFunction = mappedResult;
+                return this.proxy.$executeCommand(mappedId, ...mappedArgs).then((result: any) => {
+                    if (!Array.isArray(result) || result === undefined) {
```

Comment:
Could someone elaborate what everything what is not array should be ignored? I cannot follow why this code has to be changed. It's used extensively from many places and custom commands obviously can return something what is not array.

Reply:
Command arguments and command result currently are treated as arrays of objects. If array element is elected to be converted, according conversion function will be used to convert the type of such element. If there is no conversion function defined for an element type - such element is to stay unchanged.
Command arguments are always an array,.
Two `vscode.execute...` commands that were added always return an array of Location objects.
So, currently I don't know any commands returning not an array as result and requiring the result conversion. 
Probably this may be changed in future.

Reply:
Command results should not be treated as arrays: whether a command returns an array or a single object or even just a number is up to the command. We can construct composite conversion functions for array results in `KnownCommands`. This code here should not know anything about concrete commands.

Reply:
It seems to be hold true that monaco commands are always returning undefined (void) and exposed vscode commands are always return array or undefined. Although a type says that it can be anything: https://github.com/microsoft/vscode/blob/355eea22ea9a98a5d0517ffeae5bd176e96bb185/src/vs/workbench/api/common/extHostApiCommands.ts#L45

I'm concerned because code is not easy to grasp here. Anyway if you sure that it does not break let's go with it.